### PR TITLE
Update spaceflights tutorial and starter requirements for kedro-datasets optional dependencies

### PR DESCRIPTION
**Pull request by lrcouto**
## Description

Part of the proposal for https://github.com/kedro-org/kedro-plugins/issues/313. To be merged after `kedro-datasets 3.0.0` release. Updates the requirements for test starters and documentation related to the spaceflights tutorial to use the new naming convention for kedro-datasets optional dependencies.

Depends on https://github.com/kedro-org/kedro-plugins/pull/570 and https://github.com/kedro-org/kedro-starters/pull/212

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by lrcouto** [comment]
Tests are expected to fail until the kedro-datasets 3.0.0 release.

**Review by astrojuanlu — APPROVED** [review]
I love this!

**Comment by lrcouto** [comment]
e2e tests failing due to `ValueError: Failed to find the pipeline named '__default__'. It needs to be generated and returned by the 
'register_pipelines' function.`, but they all pass on my local environment and on a gitpod instance, so I can't reproduce it. 🤔

**Comment by merelcht** [comment]
> e2e tests failing due to `ValueError: Failed to find the pipeline named '__default__'. It needs to be generated and returned by the 'register_pipelines' function.`, but they all pass on my local environment and on a gitpod instance, so I can't reproduce it. 🤔

If you scroll up a bit you'll see: 
```
import pandas as pd                                                                    
ModuleNotFoundError: No module named 'pandas'    
```

and 
```
import numpy as np                                                                     
ModuleNotFoundError: No module named 'numpy'
``` 

so it looks like something is going wrong with installation of dependencies..

**Comment by merelcht** [comment]
ahhh I know what's going on here. `kedro-datasets` 3.0.0 only supports python 3.9+ so for the 3.8 build it can't find this new reference of `kedro-datasets[pandas-csvdataset]` and so it can't install the dependencies correctly.

File: docs/source/tutorial/tutorial_template.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ pytest~=7.2
 
 # Kedro dependencies and datasets to work with different data formats (including CSV, Excel, and Parquet)
 kedro~=0.19.0
-kedro-datasets[pandas.CSVDataset, pandas.ExcelDataset, pandas.ParquetDataset]>=1.1
+kedro-datasets[pandas-csvdataset, pandas-exceldataset, pandas-parquetdataset]>=1.1
```
Comment (merelcht) [inline docs/source/tutorial/tutorial_template.md:51]: This syntax only works for the latest datasets version:
```suggestion
kedro-datasets[pandas-csvdataset, pandas-exceldataset, pandas-parquetdataset]>=3.0
```

**Review by merelcht — APPROVED** [review]

**Pull request merged**